pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -135,6 +136,11 @@ public synchronized void init(PinotConfiguration config, HelixManager helixManag
     // used to initialize a segment refresh semaphore to limit the parallelism, so create a pool of same size.
     _segmentRefreshExecutor = Executors.newFixedThreadPool(getMaxParallelRefreshThreads(),
         new ThreadFactoryBuilder().setNameFormat("segment-refresh-thread-%d").build());
+    LOGGER.info("Initialized segment refresh executor thread pool: {}", getMaxParallelRefreshThreads());
+    _segmentPreloadExecutor = Executors.newFixedThreadPool(_instanceDataManagerConfig.getMaxSegmentPreloadThreads(),
```

Comment:
Do we always want to create this executor? Or by default not creating it (0 preload thread)? Currently the table data manager can take nullable executor

Reply:
sgtm